### DISCUSSION TITLE
MK3 Firmware detected on MK3S printer ??

### DISCUSSION THREAD

**Discussion by RogerInHawaii** [prusa3d/Prusa-Firmware#4799] [Q&A]
I've downloaded the Open Source Prusa MK3 Firmware from this GitHub repository, made some changes to it, and flashed it onto my MK3S printer. When it starts up it reports in the on-printer display that:

MK3 firmware
detected on MK3S
printer

Why is it showing that? Doesn't the firmware basically self-detect what machine it's running on and configures itself accordingly?

I don't see any repository specifically for the MK3S, but the MK3 repository SEEMS to indicate that it's applicable to the MK3S. 

Is there something within the firmware source that I need to set?

**Comment by gudnimg** [comment]
> Doesn't the firmware basically self-detect what machine it's running on and configures itself accordingly?

No, you need to compile the firmware specifically for MK3S/MK3S+. This project supports MK2.5 too not just MK3 variants.

How are you building the firmware image? I recommend using CMake if you’re not already using it.

**Comment by RogerInHawaii** [comment]
Thank you for responding.

How exactly do I do that? Is there some setting in one of the source files?

I have a Windows machine on which I have Ubuntu installed and using the Ubuntu version of Microsoft Visual Studio Code.
I compile via the Ubunto terminal using:
david@RogersBrain:~/Prusa-Firmware-MK3/build$ ninja ALL_ENGLISH
which creates the MK3_FW_3.14.1-RC1_ENGLISH.hex file that I then flash into my printer.

That may be a round-about way, but I had a friend help me get that all set up and it was the best he could do. Windows just wouldn't let us install Visual Studio Code to work under Windows, hence the need to install and use Ubuntu.

**Reply by gudnimg** [reply]
I use Windows only without any problems. I normally dont use the command line as the CMake plugin works well.  

The target you’re using `ALL_ENGLISH` is likely the issue. Usually the target name also includes which printer it is for. I’ll see how it works on my machine and reply later. 

Try checking what targets are available. `ninja -t targets all` it should have `MK3S` in its name.

**Reply by gudnimg** [reply]
@RogerInHawaii Running `ninja MK3S_ENGLISH` works on my end. This target is for stock MK3S / MK3S+

If you have a revo nozzle you will need either `MK3S-E3DREVO_ENGLISH` or `MK3S-E3DREVO_HF_60W_ENGLISH`.

**Comment by RogerInHawaii** [comment]
Thank you for taking the time to help me out. I really appreciate it.

I did as you suggested and ran ninja MK3S_English

and the hex file it produces is

MK3S_MK3S+_FW_3.14.1-RC1_ENGLISH.hex   I'll give that a try and see how it goes.

By the way, any idea what the RC1 part of that file name means?

**Comment by RogerInHawaii** [comment]
Hmmm,

I did the:

ninja MK3S_ENGLISH

as you suggested. 

It created:

MK3S_MK3S+_FW_3.14.1-RC1_ENGLISH.hex

I flashed that onto my printer.

After the flashing it showed

"Warning: printer type changed"

I rebooted the printer

And now it looks OK.
Strange that it reported that the printer type changed.
But, at least now it's working without that

"Mk3 firmware found on Mk3s printer"

So, thank you for your help with this.
